### ISSUE TITLE
AWS Lambda OpenCV Dependancy issues

### ISSUE THREAD

**Issue by DoctorSlimm** [ultralytics/ultralytics#1492]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Trying to run YOLO inside an arm64 AWS Lambda function using pytorch CPU-only and my docker wont build... keeps erroring out about psutil and other things, tried installing deps separatley and then ultralytics no deps (to make sure torch GPU is not installed by default) but no luck..

TL;DR -> I really really dont want to shave a yack (https://www.techtarget.com/whatis/definition/yak-shaving#:~:text=Yak%20shaving%20is%20programming%20lingo,to%20describe%20impediments%20to%20coding.)
and have to make a layer because I need to test my lambda Images locally cos theyre always full of bugs 

Pls

**Comment by glenn-jocher** [comment]
@DoctorSlimm try ultralytics/ultralytics:latest-arm64

See https://hub.docker.com/r/ultralytics/ultralytics/tags

**Comment by DoctorSlimm** [comment]
@glenn-jocher Hi Glenn! thanks for the quick reply, I am still getting the same error I have been dealing with all night till the morning. Here is my Dockerfile, and I am running on MacOS, and building for linux/arm64 , I have tried everything humanley possible over the last 6 hours to make it work but to no avail... Using Lambda Layers, and installing all manners of linux dependancy packages, and even building pytorch from source. the same error eventually comes through.

ERROR [ 7/12] RUN python3 -c "from ultralytics import YOLO; m = YOLO('/cache/torch/yolov8n-table-extraction/best.pt'
RUN python3 -c "from ultralytics import YOLO; m = YOLO('/cache/torch/yolov8n-table-extraction/best.pt', task='detect'); m.predict('https://ultralytics.com/images/zidane.jpg')":
#9 3.479 /usr/local/lib/python3.10/dist-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension

```Dockerfile
FROM ultralytics/ultralytics:latest-arm64

# Make cache directories
RUN mkdir /cache

# Set cache directories as environment variables
ENV TORCH_CACHE=/cache/torch \
    SENTENCE_TRANSFORMERS_HOME=/cache/torch/sentence_transformers \
    TRANSFORMERS_CACHE=/cache/huggingface/transformers

# Install packages
RUN pip3 install torch torchvision
RUN pip3 install boto3  "pinecone-client[grpc]" python-dotenv ccxt \
    && pip3 install pillow tqdm numpy scikit-learn scipy nltk sentencepiece \
    && pip3 install sentence-transformers \
    && pip3 install "transformers[torch]" \
    && pip3 install ultralytics

# Installing NLTK
RUN python3 -m nltk.downloader -d /var/lang/nltk_data punkt


# Installing YOLOv8n
RUN git clone https://huggingface.co/keremberke/yolov8n-table-extraction  /cache/torch/yolov8n-table-extraction
RUN python3 -c "from ultralytics import YOLO; m = YOLO('/cache/torch/yolov8n-table-extraction/best.pt', task='detect'); m.predict('https://ultralytics.com/images/zidane.jpg')"

# Download and save models
COPY save_model.sh ./

# Installing Sentence Transformers Models
RUN bash save_model.sh deepset "deepset/all-mpnet-base-v2-table"
RUN bash save_model.sh sentence_transformers "all-MiniLM-L6-v2"

# Installing Vanilla Transformers Tokenizer
RUN bash save_model.sh huggingface "sentence-transformers/all-MiniLM-L6-v2"

# Set production environment variable
ENV PRODUCTION=true

# Copy application files
COPY .env app.py upload_dataset.py parse_pdf.py parse_tables.py yolo_functions.py parse_article.py ./

# Set the entry
CMD ["app.lambda_handler"]
```

```Bash
docker build --file ./Dockerfile_ultra --no-cache --platform linux/arm64 -t upload_function_ultra:latest .
```

Error and build logs
`
=> [internal] load build definition from Dockerfile_ultra                                                                0.0s
 => => transferring dockerfile: 1.89kB                                                                                    0.0s
 => [internal] load .dockerignore                                                                                         0.0s
 => => transferring context: 2B                                                                                           0.0s
 => [internal] load metadata for docker.io/ultralytics/ultralytics:latest-arm64                                           0.0s
 => CACHED [ 1/12] FROM docker.io/ultralytics/ultralytics:latest-arm64                                                    0.0s
 => [internal] load build context                                                                                         0.0s
 => => transferring context: 311B                                                                                         0.0s
 => [ 2/12] RUN mkdir /cache                                                                                              0.3s
 => [ 3/12] RUN pip3 install torch torchvision                                                                            1.1s
 => [ 4/12] RUN pip3 install boto3  "pinecone-client[grpc]" python-dotenv ccxt     && pip3 install pillow tqdm numpy sc  33.6s
 => [ 5/12] RUN python3 -m nltk.downloader -d /var/lang/nltk_data punkt                                                   3.4s
 => [ 6/12] RUN git clone https://huggingface.co/keremberke/yolov8n-table-extraction  /cache/torch/yolov8n-table-extract  1.8s
 => ERROR [ 7/12] RUN python3 -c "from ultralytics import YOLO; m = YOLO('/cache/torch/yolov8n-table-extraction/best.pt'  4.7s
------
 > [ 7/12] RUN python3 -c "from ultralytics import YOLO; m = YOLO('/cache/torch/yolov8n-table-extraction/best.pt', task='detect'); m.predict('https://ultralytics.com/images/zidane.jpg')":
#9 3.479 /usr/local/lib/python3.10/dist-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension:
#9 3.479   warn(f"Failed to load image Python extension: {e}")
#9 4.049 Traceback (most recent call last):
#9 4.049   File "<string>", line 1, in <module>
#9 4.050   File "/usr/src/ultralytics/ultralytics/yolo/engine/model.py", line 96, in __init__
#9 4.051     self._load(model, task)
#9 4.051   File "/usr/src/ultralytics/ultralytics/yolo/engine/model.py", line 135, in _load
#9 4.051     self.model, self.ckpt = attempt_load_one_weight(weights)
#9 4.051   File "/usr/src/ultralytics/ultralytics/nn/tasks.py", line 395, in attempt_load_one_weight
#9 4.051     ckpt, weight = torch_safe_load(weight)  # load ckpt
#9 4.051   File "/usr/src/ultralytics/ultralytics/nn/tasks.py", line 335, in torch_safe_load
#9 4.051     return torch.load(file, map_location='cpu'), file  # load
#9 4.051   File "/usr/local/lib/python3.10/dist-packages/torch/serialization.py", line 815, in load
#9 4.051     return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
#9 4.051   File "/usr/local/lib/python3.10/dist-packages/torch/serialization.py", line 1033, in _legacy_load
#9 4.051     magic_number = pickle_module.load(f, **pickle_load_args)
#9 4.051 _pickle.UnpicklingError: invalid load key, 'v'.
------
executor failed running [/bin/sh -c python3 -c "from ultralytics import YOLO; m = YOLO('/cache/torch/yolov8n-table-extraction/best.pt', task='detect'); m.predict('https://ultralytics.com/images/zidane.jpg')"]: exit code: 1
`

**Comment by glenn-jocher** [comment]
@DoctorSlimm if this works for an official model, i.e. `model=YOLO('yolov8n.pt')` then the custom model you are using is simply corrupt or out of date or problematic in some way.

**Comment by DoctorSlimm** [comment]
@glenn-jocher thanks! getting back into this over the weekend, will try to make it work on https://www.banana.dev/ Serverless GPU Containers :)

**Comment by glenn-jocher** [comment]
@DoctorSlimm great! Please submit a PR if there are any changes here that might make life easier for lambda functions.
